File: back-end/src/main/java/com/PocketPilot/project/propositionReduction/ResumeProposition.java
package com.PocketPilot.project.propositionReduction;

import java.util.List;

@lombok.Data
@lombok.Builder
@lombok.NoArgsConstructor
@lombok.AllArgsConstructor
public class ResumeProposition {
    private int nombreCategoriesConcernees;
    private java.math.BigDecimal gainMensuelTotal;
    private java.math.BigDecimal epargneMensuelleNecessaire;
    private Boolean objectifAtteignable;
    private Integer nombreMoisNecessaires;
    private String commentaire;
    private List<PropositionReduction> propositions;
}

File: front-end/src/types/resumeProposition.ts
import type { PropositionReduction } from "./propositionReduction "

export interface ResumeProposition {
  nombreCategoriesConcernees: number
  gainMensuelTotal: number
  epargneMensuelleNecessaire: number
  objectifAtteignable: boolean
  nombreMoisNecessaires: number | null
  commentaire: string | null
  propositions: PropositionReduction[]
}